examples/hello_compute_c/main.c

```diff
@@ -0,0 +1,164 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#define WGPU_TARGET_MACOS 1
+#define WGPU_TARGET_LINUX 2
+#define WGPU_TARGET_WINDOWS 3
+
+#define BLEND_STATES_LENGTH (1)
+#define ATTACHMENTS_LENGTH (1)
+#define RENDER_PASS_ATTACHMENTS_LENGTH (1)
+#define BINDINGS_LENGTH (1)
+#define BIND_GROUP_LAYOUTS_LENGTH (1)
+
+WGPUByteArray read_file(const char *name) {
+    FILE *file = fopen(name, "rb");
+    fseek(file, 0, SEEK_END);
+    long length = ftell(file);
+    unsigned char *bytes = malloc(length);
+    fseek(file, 0, SEEK_SET);
+    fread(bytes, 1, length, file);
+    fclose(file);
+    return (WGPUByteArray){
+        .bytes = bytes,
+        .length = length,
+    };
+}
+
+void read_buffer_map(
+    WGPUBufferMapAsyncStatus status, const uint8_t *data, uint8_t *userdata) {
+    if (status == WGPUBufferMapAsyncStatus_Success) {
+        uint32_t *times = (uint32_t *)data;
+        printf("Times: [%d, %d, %d, %d]", times[0], times[1], times[2], times[3]);
+    }
+}
+
+int main(int argc, char *argv[]) {
+
+    if (argc != 5) {
+        return 0;
+    }
+
+    uint32_t numbers[] = {
+        strtoul(argv[1], NULL, 0),
+        strtoul(argv[2], NULL, 0),
+        strtoul(argv[3], NULL, 0),
+        strtoul(argv[4], NULL, 0),
+    };
+
+    int size = sizeof(numbers);
+
+    uint32_t numbers_length = size / sizeof(uint32_t);
+
+    WGPUInstanceId instance = wgpu_create_instance();
+
+    WGPUAdapterId adapter = wgpu_instance_get_adapter(instance,
+        &(WGPUAdapterDescriptor){
+            .power_preference = WGPUPowerPreference_LowPower,
+        });
+
+    WGPUDeviceId device = wgpu_adapter_request_device(adapter,
+        &(WGPUDeviceDescriptor){
+            .extensions = {
+				.anisotropic_filtering = false,
+            },
+        });
+
+	uint8_t *staging_memory;
+
+    WGPUBufferId staging_buffer = wgpu_device_create_buffer_mapped(device,
+        &(WGPUBufferDescriptor){
+			.size = size,
+            .usage = WGPUBufferUsage_MAP_READ | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC},
+            &staging_memory);
+
+	memcpy((uint32_t *)staging_memory, numbers, size);
+	
+	wgpu_buffer_unmap(staging_buffer);	
+
+    WGPUBufferId storage_buffer = wgpu_device_create_buffer(device,
+        &(WGPUBufferDescriptor){
+			.size = size,
+            .usage = WGPUBufferUsage_STORAGE | WGPUBufferUsage_TRANSFER_DST |
+                WGPUBufferUsage_TRANSFER_SRC});
+
+    WGPUBindGroupLayoutId bind_group_layout =
+        wgpu_device_create_bind_group_layout(device,
+            &(WGPUBindGroupLayoutDescriptor){
+                .bindings = &(WGPUBindGroupLayoutBinding){
+					.binding = 0,
+                    .visibility = WGPUShaderStage_COMPUTE,
+                    .ty = WGPUBindingType_StorageBuffer},
+				.bindings_length = BINDINGS_LENGTH});
+
+	WGPUBindingResource resource = {
+        .tag = WGPUBindingResource_Buffer,
+        .buffer = (WGPUBindingResource_WGPUBuffer_Body) {
+            storage_buffer
```

Comment:
this needs to be a struct:
```rs
typedef struct {
  WGPUBufferId buffer;
  WGPUBufferAddress offset;
  WGPUBufferAddress size;
} WGPUBufferBinding;
```

We are missing the size specified. In the future, the default will be here to assume the whole buffer, but today the size is still needed.

Reply:
@Napokue I think technically we are supposed to have both `WGPUBindingResource_WGPUBuffer_Body` and `WGPUBufferBinding` here. The binding resource (`WGPUBindingResource`) accepts a `buffer` of type `WGPUBindingResource_WGPUBuffer_Body`.

The binding (`WGPUBufferBinding`) is a field on the buffer binding resource (`WGPUBindingResource_WGPUBuffer_Body`), see https://github.com/gfx-rs/wgpu/blob/32399cff8a0b55389f2d2493ea5c900c986c2547/ffi/wgpu.h#L374-L376

Please let me know if this isn't clear. We do want to improve this API in the future (#219)